File: frontend/src/api/reportes.ts
import api from '@/lib/api'

export interface ConsumoAreaRow {
  area_id: number
  area_nombre: string
  mes: string
  total_consumido: number
  unidades_distintas: number
  movimientos_count: number
}

export interface TopDescartadoRow {
  producto_id: string
  producto_nombre: string
  total_descartado: number
  unidad: string
  movimientos_count: number
}

export interface ReporteParams {
  desde?: string
  hasta?: string
  limit?: number
}

export const reportesApi = {
  consumoArea: (params: ReporteParams) =>
    api.get<ConsumoAreaRow[]>('/reportes/consumo-area', { params }).then((r) => r.data),

  topDescartados: (params: ReporteParams) =>
    api.get<TopDescartadoRow[]>('/reportes/top-descartados', { params }).then((r) => r.data),
}

File: backend/src/dto/reportes.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct ReporteParams {
    pub desde: Option<String>,
    pub hasta: Option<String>,
    pub limit: Option<i64>,
}

#[derive(Debug, Serialize, sqlx::FromRow)]
pub struct ConsumoAreaRow {
    pub area_id: i32,
    pub area_nombre: String,
    pub mes: String,
    pub total_consumido: f64,
    pub unidades_distintas: i64,
    pub movimientos_count: i64,
}

#[derive(Debug, Serialize, sqlx::FromRow)]
pub struct TopDescartadoRow {
    pub producto_id: String,
    pub producto_nombre: String,
    pub total_descartado: f64,
    pub unidad: String,
    pub movimientos_count: i64,
}
